temporal-opentracing/src/main/java/io/temporal/opentracing/internal/SpanFactory.java

```diff
@@ -77,6 +77,46 @@ public Tracer.SpanBuilder createChildWorkflowStartSpan(
     return createSpan(context, tracer, null, References.CHILD_OF);
   }
 
+  public Tracer.SpanBuilder createExternalWorkflowSignalSpan(
+      Tracer tracer, String signalName, String workflowId, String runId) {
+    SpanCreationContext context =
+        SpanCreationContext.newBuilder()
+            .setSpanOperationType(SpanOperationType.SIGNAL_EXTERNAL_WORKFLOW)
+            .setActionName(signalName)
+            .setWorkflowId(workflowId)
+            .setRunId(runId)
+            .build();
+    return createSpan(context, tracer, null, References.CHILD_OF);
+  }
+
+  public Tracer.SpanBuilder createWorkflowSignalSpan(
+      Tracer tracer, String signalName, String workflowId, String runId) {
+    SpanCreationContext context =
+        SpanCreationContext.newBuilder()
+            .setSpanOperationType(SpanOperationType.SIGNAL_WORKFLOW)
+            .setActionName(signalName)
+            .setWorkflowId(workflowId)
+            .setRunId(runId)
+            .build();
+    return createSpan(context, tracer, null, References.FOLLOWS_FROM);
+  }
+
+  public Tracer.SpanBuilder createWorkflowSignalSpan(
```

Comment:
```suggestion
  public Tracer.SpanBuilder createWorkflowHandleSignalSpan(
```

Can you change the names for handle-based spans instead of overloads where the caller can't know they are creating different operation types just by calling different overloads?

Reply:
done